delivery_driver/models/res_partner.py

```diff
@@ -16,7 +16,9 @@ def _check_is_driver(self):
         drivers_in_carrier = self.env["delivery.carrier"].search(
             [("driver_id", "in", self.ids)]
         )
-        if drivers_in_carrier and not self.is_driver:
+        if drivers_in_carrier and drivers_in_carrier.driver_id.filtered_domain(
+            [("is_driver", "=", False)]
```

Comment:
Will fail if drivers_in_carriers are more than one.

```suggestion
        if drivers_in_carrier and drivers_in_carrier.filtered_domain(
            [("driver_id.is_driver", "=", False)]
```